### PR TITLE
cleanup move_pr.sh to make it suitable for CI for v2

### LINKED ISSUE #13651: [Bug][move-compiler] move_pr.sh script is broken if git-nextest is an old version
# 🐛 Bug

It usually doesn't seem to actually run any tests.

### PR DESCRIPTION
**Pull request by brmataptos**
## Description

Cleanup of the the move_pr.sh script to allow use in CI:
- relocate v2 doc test outputs, use v2 to build for v2 integration tests
- use `nextest -profile smoke-test` by default to tolerate rare Heisenbugs by replaying 3x
- a few cleanups

Slightly more description:
- Small cleanups to `move_pr.sh`, including actually paying attention to `MOVE_PR_PROFILE` env var.
- Use `nextest --profile smoke-test` by default to allow re-runs in case of intermittent failures (Heisenbugs).  Use env var `MOVE_PR_NEXTEST_PROFILE` to allow this to be overridden.
- Use new `MVC_DOCGEN_OUTPUT_DIR` flag to move tests to be used with `MOVE_COMPILER_V2=true` to relocate all move-compiler-v2-generated docs to keep them distinct from v1-generated docs.  Check in these v2 docs for comparison.
- Clean up a few compiler-v2 verification tests and v1.matched that were left over from #13732.
- Add a comment to script tell people what to do if tests aren't running due to an old version of cargo-nextest.

Fix #13651.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

run repeatedly.

## Key Areas to Review

Do we need to update `.config/nextest.toml` to force `cargo-profile = "ci"` for `[profile.ci]`?

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/scripts/move_pr.sh [hunk 1]
```diff
@@ -143,31 +143,32 @@ if [ ! -z "$CHECK" ]; then
   )
 fi
 
+CARGO_OP_PARAMS="--locked --profile $MOVE_PR_PROFILE"
```
Comment (wrwg) [inline third_party/move/scripts/move_pr.sh:146]: That does not make sense, the parameter is definitely `--cargo-profile` to select the _cargo_ profile not the nexttest one. It is essential that we use the same profile for cargo build and nextest, which seems to be hampered by this PR. Maybe by accident there is still a nextest profile of the same name, but that is not what we want to select here.

See also here: https://nexte.st/docs/running/#options-and-arguments

Why should those change be necessary in the first place?

Comment (brmataptos) [reply third_party/move/scripts/move_pr.sh:149]: As discussed off-line, this was needed due to an outdated version of `carg-nextest`.  I've updated this PR to add a comment to warn about this case and provide a solution, and also correctly read the `MOVE_PR_PROFILE` environment variable if it is defined.

**Review by vineethk — APPROVED** [review]
LGTM.

**Comment by brmataptos** [comment]
Ok, I've changed docgen to add a `MVC_DOCGEN_OUTPUT_DIR` env var (for testing only) to put the V2-generated `.md` files in a special dir `tests/compiler-v2-doc` for this test, allowing safe coexistence with V1.  I've made it suitable for CI, I think.

**Comment by brmataptos** [comment]
PTAL

**Review by wrwg — APPROVED** [review]

**Review by vineethk — APPROVED** [review]

**Pull request merged**